### PR TITLE
Update requirements.txt

### LINKED ISSUE #1: Add InsightIDR Backend
Hello! Love the project, I will definitely be a user and promoter. 😄 

I suggest adding the Rapid7 InsightIDR backend to the project: https://github.com/SigmaHQ/pySigma-backend-insightidr

It is compatible with the latest version of pySigma (it will work fine with pySigma 9.x as well).

### PR DESCRIPTION
**Pull request by nasbench**
This PR add the following:

- Adds InsightIDR and closes #1 
- Update some pipelines and backends to their latest version

### CODE REVIEW

**Comment by josehelps** [comment]
hey @nasbench can you update https://github.com/magicsword-io/sigconverter.io/blob/main/pyproject.toml as well keep them insyc.

**Review by josehelps — APPROVED** [review]
LGTM

**Review by MHaggis — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -16,16 +16,17 @@ sigma-cli = "0.7.6"
...
 pysigma-backend-qradar = "0.3.3"
 pysigma-backend-qradar-aql = "0.2.3"
 pysigma-backend-sentinelone = "0.1.2"
 pysigma-backend-sentinelone-pq = "0.1.1"
-pysigma-backend-splunk = "1.0.2"
-pysigma-pipeline-crowdstrike = "1.0.0"
-pysigma-pipeline-sysmon = "1.0.2"
-pysigma-pipeline-windows = "1.1.0"
+pysigma-backend-splunk = "1.0.3"
+pysigma-backend-insightidr=="0.2.2"
```
Comment (M3NIX) [inline pyproject.toml:26]: @nasbench there is an extra equal which is breaking poetry

**Comment by M3NIX** [comment]
@nasbench @MHaggis 
I think we should revert this pull request because the updated backends/pipelines each depend on different pysigma versions. therefore we can not simply update all backends at the same time because with the current implementation we can have only one pysigma version installed.

when you try to install the dependencies with pip or poetry both will return an error:
`pip` output:
```
ERROR: Cannot install -r requirements.txt (line 2), -r requirements.txt (line 4), -r requirements.txt (line 5), -r requirements.txt (line 6) and pysigma==0.9.11 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested pysigma==0.9.11
    sigma-cli 0.7.6 depends on pysigma<0.10.0 and >=0.9.9
    pysigma-backend-carbonblack 0.1.4 depends on pysigma<0.10.0 and >=0.9.6
    pysigma-backend-cortexxdr 0.1.1 depends on pysigma<0.10.0 and >=0.9.8
    pysigma-backend-elasticsearch 1.0.6 depends on pysigma<0.11.0 and >=0.10.2

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict
```

`poetry` output:
```
$> poetry install
Updating dependencies
Resolving dependencies... (4.7s)

Because pysigma-pipeline-windows (1.1.1) depends on pysigma (>=0.10.2,<0.11.0)
 and sigconverter-io depends on pysigma (0.9.11), pysigma-pipeline-windows is forbidden.
So, because sigconverter-io depends on pysigma-pipeline-windows (1.1.1), version solving failed.
```

**Comment by nasbench** [comment]
Yeah that's a known issue with the Pysigma plugin system. Was making a test for it. We'll deploy a fix.

**Pull request merged**